Add initial system design for unscented transform

adamlm opened the pull request:
<!-- Thanks for the contribution, this is awesome. -->

# PR Details
## Description

<!--- Describe your changes in detail -->

This PR adds a high level description and initial API design for the unscented transform (UT). The UT is a key component in propagating detected objects' states and state covariances.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

N/A

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

Writing documentation can help reveal implementation challenges and improve our understanding of what we're actually trying develop.

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

N/A

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Defect fix (non-breaking change that fixes an issue)
- [] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)
- [x] Additional documentation

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

adamlm committed:
Add system design for unscented transform

State and covariance prediction will be done using an unscented transform, so
we should provide a brief overview about it and describe the associated API
design.

docs/system_design/unscented_transform.rst
* add unscented transform overview
* add API design overview

drpjm commented on docs/system_design/unscented_transform.rst:
```diff
@@ -0,0 +1,106 @@
+
+Sigma points represents a sampling over the object state's distribution. We could sample the distribution (*i.e.*,
+generate sigma points) using one of many methods. We need to generate at least :math:`n + 1` points, where :math:`n` is
+the number of state variables. Generating :math:`2n + 1` point is common.
```
Tweak to say "...points is a common heuristic."

drpjm commented on docs/system_design/unscented_transform.rst:
```diff
@@ -0,0 +1,106 @@
+        s_{i} &= \boldsymbol{x}_{k} - \sqrt{n \boldsymbol{P}_{k, [i - n]}} \quad \text{for } i = n, \dots, 2n + 1 \\
+    \end{align}
+
+Notation :math:`\boldsymbol{x}_{k}` and :math:`\boldsymbol{P}_{k}` represents the object's state vector
```
Mean of the state instead?

drpjm reviewed (changes requested):
Just a couple language tweaks requested. Otherwise good to go.

adamlm committed:
Update text in unscented transform docs

Update and improve some wording in the unscented transform docs based on PR
reviews.

docs/system_design/unscented_transform.rst
* add that sigma point generation is a heuristic
* add that detected object's state vector is the mean of the distribution
